sycl/include/sycl/reduction.hpp

```diff
@@ -1014,7 +1014,7 @@ class reduction_impl_algo {
             }
           });
         } else {
-          accessor OutAcc{Out, CGH};
+          accessor OutAcc{Out, CopyHandler};
```

Comment:
Sorry, but I don't seem to understand why this change was made? Can you please elaborate?

Reply:
@uditagarwal97 , good point, it might deserve an extended comment in the commit message.

In old behavior, `OutAcc` was connected with `CGH`, but this is not enough for the correct operations, so later same accessor was connect with `CopyHandler` via `addAccessorReq()`. But connection with `CGH` is not required, thus allowing us to connect with `CopyHandler` right away and eliminate `addAccessorReq()` calls.